### PR TITLE
feat(shared): change the winjs font for a new fluent style font

### PR DESCRIPTION
**Pull request by alextrepa**
GitHub Issue (If applicable): [#3011](https://github.com/unoplatform/uno/issues/3011), [967](https://github.com/unoplatform/uno/issues/967)

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?
We currently use the Winjs-symbols font. Since it is not and exact match for Segoe there are control symbols that are missing.

## What is the new behavior?

This is very much a work in progress. It is a first draft of a font that will be used as a default for Uno controls. It takes most of its symbols from the [Fluent System Icons](https://github.com/microsoft/fluentui-system-icons). As a fall back, we use Icons from [Icons 8 Line awesome](https://icons8.com/line-awesome) and Winjs. Thank you to @kakone for his help as he had already did part of the icon mapping and made a pretty great tool that was very useful (https://github.com/kakone/SvgFont2Files).

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

This PR will bring some breaking change. For now it is a draft. I will add further details.

## Other information

# Uno Fluent UI assets
Uno has a new multiplateform font. The new font must be added to app build before the change.

> [!IMPORTANT] \
This is a breaking change on most platform the font needs to be changed or most control will display incorrectly.

## Font files
The updated fonts are here.
[uno-fluentui-assets.zip](https://github.com/unoplatform/uno/files/5058390/uno-fluentui-assets.zip)

## Changes
| Device family | Description   |
| -- | -- |
| iOS & macOS | Once Uno has been updated, it will start looking for a font named symbols (Not necessarily the name of the file). For this font to be available it needs to be in the resource folder.  image image |
| Android | Once Uno has been updated, it will start looking for a font file named uno-fluentui-assets.ttf in its assets folder:  image |
| WebAssembly | A wasm build won't break after the update but to access the new symbols the file Font.css should be chnaged. The font is passed as a base64 file. image |

## Limitations
On iOS and macOS the indeterminate state for a check box is not the right color.

## Related Topics
- [#3011](https://github.com/unoplatform/uno/issues/3011)
- [967](https://github.com/unoplatform/uno/issues/967)

### CODE REVIEW

File: src/SamplesApp/SamplesApp.iOS/SamplesApp.iOS.csproj [hunk 1]
```diff
@@ -107,9 +107,7 @@
     <None Include="Info.plist" />
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Content Include="Entitlements.plist" />
-    <BundleResource Include="Resources\Fonts\winjs-symbols.ttf">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (davidjohnoliver) [inline src/SamplesApp/SamplesApp.iOS/SamplesApp.iOS.csproj:111]: I think you should leave this line there, unless someone can confirm that it's not necessary.

File: src/SamplesApp/SamplesApp.macOS/SamplesApp.macOS.csproj [hunk 2]
```diff
@@ -140,44 +140,34 @@
...
     <None Include="Entitlements.plist" />
+    <None Include="Resources\Fonts\uno-fluentui-assets.ttf" />
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Main.cs" />
     <Compile Include="AppDelegate.cs" />
   </ItemGroup>
-  <ItemGroup>
-    <BundleResource Include="Resources\Fonts\winjs-symbols.ttf">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (davidjohnoliver) [inline src/SamplesApp/SamplesApp.macOS/SamplesApp.macOS.csproj:158]: same here

File: src/SamplesApp/SamplesApp.macOS/SamplesApp.macOS.csproj [hunk 3]
```diff
@@ -140,44 +140,34 @@
     <ImageAsset Include="Assets.xcassets\unologo.imageset\Contents.json" />
     <ImageAsset Include="Assets.xcassets\unologo.imageset\unoplatform.jpg" />
   </ItemGroup>
-  <ItemGroup>
-    <Folder Include="Resources\" />
-    <Folder Include="Resources\Fonts\" />
-    <Folder Include="Assets.xcassets\unologo.imageset\" />
-  </ItemGroup>
```
Comment (davidjohnoliver) [inline src/SamplesApp/SamplesApp.macOS/SamplesApp.macOS.csproj:147]: did you mean to remove these?

File: src/SamplesApp/SamplesApp.macOS/SamplesApp.macOS.csproj (hunk 2 above)
Comment (davidjohnoliver) [reply src/SamplesApp/SamplesApp.macOS/SamplesApp.macOS.csproj:158]: actually in this case it's worse, seems like the `BundleResource` build action got removed

File: src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/Uno.ProjectTemplates.Dotnet.csproj [hunk 4]
```diff
@@ -80,6 +80,18 @@
...
+	  <None Remove="content\unoapp-prism\BlankApp.iOS\Resources\Fonts\uno-fluentui-assets.ttf" />
+	  <None Remove="content\unoapp-prism\BlankApp.macOS\Resources\Fonts\uno-fluentui-assets.ttf" />
+	</ItemGroup>
+	
+	<ItemGroup>
+	  <UpToDateCheckInput Remove="content\unoapp-prism\BlankApp.Droid\Assets\Fonts\uno-fluentui-assets.ttf" />
+	  <UpToDateCheckInput Remove="content\unoapp-prism\BlankApp.iOS\Resources\Fonts\uno-fluentui-assets.ttf" />
+	  <UpToDateCheckInput Remove="content\unoapp-prism\BlankApp.macOS\Resources\Fonts\uno-fluentui-assets.ttf" />
+	</ItemGroup>
+
```
Comment (davidjohnoliver) [inline src/SolutionTemplate/Uno.ProjectTemplates.Dotnet/Uno.ProjectTemplates.Dotnet.csproj:94]: why do you need these?

File: src/SolutionTemplate/UnoSolutionTemplate.VISX/UnoSolutionTemplate.VISX.csproj [hunk 5]
```diff
@@ -83,9 +83,7 @@
     <None Include="source.extension.vsixmanifest">
       <SubType>Designer</SubType>
     </None>
-    <None Include="winjs-symbols.ttf">
-      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (davidjohnoliver) [inline src/SolutionTemplate/UnoSolutionTemplate.VISX/UnoSolutionTemplate.VISX.csproj:87]: same comment here, I would preserve the `CopyToOutputDirectory` attribute to be on the safe side

File: src/SamplesApp/SamplesApp.iOS/SamplesApp.iOS.csproj (hunk 1 above)
Comment (alextrepa) [reply src/SamplesApp/SamplesApp.iOS/SamplesApp.iOS.csproj:111]: I am removing all dependency on winjs-symbols. So why would we still need this?

Comment (davidjohnoliver) [reply src/SamplesApp/SamplesApp.iOS/SamplesApp.iOS.csproj:111]: what I mean is the `<CopyToOutputDirectory>Always</CopyToOutputDirectory>` line. I don't know what it does - but that's why I wouldn't remove it.

File: src/SamplesApp/SamplesApp.macOS/SamplesApp.macOS.csproj (hunk 3 above)
Comment (alextrepa) [reply src/SamplesApp/SamplesApp.macOS/SamplesApp.macOS.csproj:147]: no, I'll look into it

File: src/SamplesApp/SamplesApp.iOS/SamplesApp.iOS.csproj (hunk 1 above)
Comment (jeromelaban) [reply src/SamplesApp/SamplesApp.iOS/SamplesApp.iOS.csproj:111]: It's actually required for the bundling of fonts to work (or at least it used to).

**Comment by jeromelaban** [comment]
Make sure mention in the commit `BREAKING CHANGE:` saying the font is now changed and the newer one should be used instead.

**Comment by robloo** [comment]
Awesome work!!!

**Comment by jeromelaban** [comment]
@mergifyio backport release/beta/3.0

**Pull request merged**